provider/lxd/credentials.go

```diff
@@ -219,36 +264,122 @@ func (p environProviderCredentials) finalizeCertificateCredential(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	if !isLocalEndpoint {
-		// The endpoint is not local, so we cannot generate a
-		// certificate credential.
-		//
-		// TODO(axw) we could look in the $HOME/.config/lxc/servercerts
-		// directory for a server certificate. We would need to read
-		// $HOME/.config/lxc/config.yml to identify the remote by its
-		// endpoint.
-		//
-		// TODO(axw) for the "interactive" auth-type, we should take
-		// the user through the server certificate fingerprint
-		// verification and trust password flow.
-		return nil, errors.Errorf(`cannot auto-generate credential for remote LXD
-
-Until support is added for verifying and authenticating to remote LXD hosts,
-you must generate the credential on the LXD host, and add the credential to
-this client using "juju add-credential localhost".
-
-See: https://jujucharms.com/docs/stable/clouds-LXD
-`)
+
+	// If the end point is local, set up the local server and automate the local
+	// certificate credentials.
+	if isLocalEndpoint {
+		svr, err := p.serverFactory.LocalServer()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		cred, err := p.finalizeLocalCertificateCredential(
+			stderr, svr, certPEM, keyPEM,
+			args.Credential.Label,
+		)
+		return cred, errors.Trace(err)
+	}
+
+	// We're not local, so setup the remote server and automate the remote
+	// certificate credentials.
+	return p.finalizeRemoteCertificateCredential(
+		stderr,
+		args.CloudEndpoint,
+		args.Credential,
+	)
+}
+
+func (p environProviderCredentials) finalizeRemoteCertificateCredential(
```

Comment:
Some of this looks similar to logic in container/lxd/certificate.go and _CreateClientCertificate_ in container/lxd/server.go.

If any of that code is now unused, we can delete it.

Reply:
OK, I'll have a look.

Reply:
fixed in https://github.com/juju/juju/pull/8938/commits/4edf5be9c5ec7206586fba1c3f22a0848d74f50e